packages/composites/market-status/src/adapter.ts

```diff
@@ -0,0 +1,44 @@
+import { AdapterRequest, AdapterResponse, Execute, InputParameters } from '@chainlink/ea-bootstrap'
+import { Validator } from '@chainlink/ea-bootstrap'
```

Comment:
I believe these are from the v2 framework? We should be using v3 framework now?

Reply:
@martin-cll You're right. Looks like this is created as v2 EA. I would also suggest to create a v3 EAs using ea-framework as v2 is deprecated and we don't support it or fix bugs there.

Reply:
Hey @mxiao-cll @karen-stepanyan I have updated this PR to use the v3 EA. For reference, the reason I did not opt to do so in the first place (sorry I did not include this in the PR description) is that the v3 transports (e.g. http, ws) have a strong assumption that there is only 1 upstream data provider. This makes sense for regular adapters, but maybe less so for composite adapters such as this one which uses multiple underlying providers (i.e. other adapters).

The simplest way I could see to use the existing transports without implementing a new one was to use the http transport with a modified `Requester`. The modification is to intercept the `axios.request` call with a modified function (see `overrideAxiosRequest`) that calls the multiple underlying adapters. Unfortunately, because most of the internal impl of the `Requester` class is declared `private`, I've had to copy paste the entire class to make this work.

This isn't a great final solution, but I felt that it was better than implementing a transport class from scratch in which case I would've lost out on all the goodies that both the HTTP transport and requester provided. Perhaps a future version of the framework could redeclare the Requester's internals as `protected` instead of make inheritance easier?

Reply:
hey @martin-cll . Thank you very much for detailed explanation. Actually there is no need to use workarounds with `Requester` or `axios` as the ea-framework provides all the components, tho I agree that we might need more comprehensive documentation and guides. 
You are absolutely right that `http` and `ws` transports are not a good fit here, but for any non standard data fetching/processing, you can create a custom transport. We have many examples in monorepo of custom transport EAs, and we actually have a custom composite transport (gm-token) that uses 3 different source EAs - https://github.com/smartcontractkit/external-adapters-js/blob/main/packages/composites/gm-token/src/transport/price.ts#L139 in its logic. 
When creating a new EA with `yarn new` you can choose `Custom (subscription based)` which will generate the most of boilerplate code. I think it will be much simpler if we use custom, subscription based transport approach for this EA. Let me know if you have concerns or questions and we can go through this privately

Reply:
@karen-stepanyan Makes sense, thanks for showing me the custom transport! I have updated this PR to use a similar transport.